src/metrics/enhanced-metrics.ts

```diff
@@ -49,12 +49,17 @@ export function getRuntimeTag(): string | null {
   return `runtime:${processVersionTagString}`;
 }
 
-export function getEnhancedMetricTags(context: Context): string[] {
-  let arnTags = [`functionname:${context.functionName}`];
-  if (context.invokedFunctionArn) {
-    arnTags = parseTagsFromARN(context.invokedFunctionArn, context.functionVersion);
+export function getEnhancedMetricTags(context?: Context): string[] {
+  const tags: string[] = [];
+  if (context) {
+    let arnTags = [`functionname:${context.functionName}`];
+    if (context.invokedFunctionArn) {
```

Comment:
I think it's guaranteed that the `context.invokeFunctionArn` [is going to be present](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html).

Do we really need the condition?

Reply:
Thanks for clarification